File: C#OOP/Encapsulation and Polymorphism/Shapes/Circle.cs
using System;

namespace Shapes
{
    public class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius
        {
            get { return this.radius; }

            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("Radius", "Radius can't be 0 or less");
                }

                this.radius = value;
            }
        }
        public double CalculateArea()
        {
            double area = Math.PI * radius * radius;
            return area;
        }

        public double CalculatePerimeter()
        {
            double perimeter = 2 * Math.PI * radius;
            return perimeter;
        }
    }
}

File: JavaOOP/Geometry/src/Geometry2D/Circle.java
package Geometry2D;


public class Circle extends PlaneShape {
	private double radius;
	public Circle(Vertex2D[] vertices, double radius) {
		super(vertices);
		this.radius = radius;
	}

	@Override
	public double calculatePerimeter() {
		double perimeter =  2 * Math.PI * this.radius;
		return perimeter;
	}

	@Override
	public double calculateArea() {
		double area = Math.PI * this.radius * this.radius;
		return area;
	}
	
	@Override
	public String toString() { 
		double perimeter = this.calculatePerimeter();
		double area = this.calculateArea();
	    return "Circle: perimeter = " + perimeter + ", area = " + area;
	}
}
